### PR TITLE
Add rule `require-throws`

### PR DESCRIPTION
**Pull request by hbergren**
This adds a new rule `require-throws`, that requires a `@throws` statement in the JSDoc if the function body contains the `throw` keyword.

I based this off of `require-returns`, as suggested in https://github.com/gajus/eslint-plugin-jsdoc/issues/566

### CODE REVIEW

File: src/rules/requireThrows.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  utils,
+}) => {
+  // A preflight check. We do not need to run a deep check for abstract functions.
+  if (canSkip(utils)) {
+    return;
+  }
+
+  const tagName = utils.getPreferredTagName({tagName: 'throws'});
+  if (!tagName) {
+    return;
```
[the remark this answers is not in the document]
Comment (brettz9) [reply src/rules/requireThrows.js:36]: Should be enough to add a (passing) example which doesn't have `@throws` at all (or where it has `@throws` but `settings.jsdoc.tagNamePreference` is set to map `throws` to something else like `throw` and if `@throw` wasn't present).

**Comment by brettz9** [comment]
Looks very good from a quick look!

File: src/rules/requireThrows.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (hbergren) [reply src/rules/requireThrows.js:36]: This is the _only_ line that I can't figure out how to cover. I feel like it should be covered by [this test](https://github.com/gajus/eslint-plugin-jsdoc/pull/574/files#diff-2a64399eb55f3de7b95b87088f3bc8b8R324), but it seems like I'm not understanding how `utils.getPreferredTagName()` is working?

[the remark this answers is not in the document]
Comment (brettz9) [reply src/rules/requireThrows.js:36]: No worries, I've added the test (it was setting `throws` to `false` on `tagNamePreference` settings, i.e., if someone had indicated they wished to block the `throws` tag yet were still using this rule. :)

I think it looked good, but need a little more time to give it another look over before merging.

**Comment by hbergren** [comment]
@brettz9 , thanks for adding the missing test! I'll know for next time to add a case where the tag is given in the JSDoc but should not be.

**Comment by brettz9** [comment]
Great work, thanks!

**Pull request merged**